Feature/core/FriendService

Anakael opened the pull request.

Anakael committed:
Remove exceeds links

Anakael committed:
Add frinend service and route

Merge branch 'release/0.1.0' of github.com:Anakael/Treasury-Department into feature/core/friendservice

# Conflicts:
#	TreasuryDepartment/Models/BlackList.cs
#	TreasuryDepartment/Models/Friend.cs
#	TreasuryDepartment/Models/TreasuryDepartmentContext.cs

Anakael committed:
Add frinend service and route

Merge branch 'release/0.1.0' of github.com:Anakael/Treasury-Department into feature/core/friendservice

# Conflicts:
#	TreasuryDepartment/Models/BlackList.cs
#	TreasuryDepartment/Models/Friend.cs
#	TreasuryDepartment/Models/TreasuryDepartmentContext.cs

Anakael committed:
SonarQube analysis

Add frinend service and route

Merge branch 'release/0.1.0' of github.com:Anakael/Treasury-Department into feature/core/friendservice

# Conflicts:
#	TreasuryDepartment/Models/BlackList.cs
#	TreasuryDepartment/Models/Friend.cs
#	TreasuryDepartment/Models/TreasuryDepartmentContext.cs

arlekin147 commented on TreasuryDepartment/Controllers/UsersController.cs:
```diff
@@ -1,46 +1,37 @@
+    [ApiController]
+    public class UsersController : ControllerBase
+    {
+        private readonly UserService _userService;
```
Не стоит завязываться на реализацию

Anakael replied:
То есть?

arlekin147 replied:
Инъектировать интерфейс

Anakael replied:
Не вижу ситуаций, в который у нас может появиться другая реализация UserService.

arlekin147 replied:
Пока не видишь, а потом могут  появиться. И отвязка от реализации ( которая делается невеликим трудом сейчас ), может принести свои плоды. Вообще не завязываться на реализацию одно из главных правил ООП

Anakael replied:
Это и потом будет сделать не очень трудно, заменив объект на интерфейс.
У нас этот сервис достаёт пользователя из БД. Откуда мы еще его можем достать?

arlekin147 replied:
Из любого другого источника

Anakael replied:
Done?

arlekin147 commented on TreasuryDepartment/Controllers/UsersController.cs:
```diff
@@ -1,46 +1,37 @@
+        [HttpPost]
+        public async Task<ActionResult<User>> Post(User user)
+        {
+            await _userService.Create(user);
```
Сервис изменяет юзера?
Либо стоит сохранить результат метода и работать с ним, либо явно укажи, что метод будет его изменять написав: Create(ref user)

Anakael replied:
Он его создает. Метод Post же.
И это не сервис, а контроллер. Как вариант, чтобы не возникало вопросов типа "откуда берется User" - можно добавить [FromBody]

arlekin147 replied:
Я не про это. ТЫ передаешь в userService объект юзера и не получаешь результат

Anakael replied:
Да. Потому что он здесь не нужен, потому что здесь запись в базу происходит. На 34ой строке возращает редирект на метод получения пользователя.
Пример: https://docs.microsoft.com/ru-ru/aspnet/core/tutorials/first-web-api?view=aspnetcore-2.2&tabs=visual-studio#add-a-controller

arlekin147 replied:
Они там делают не так. Они получают результат сервиса и возвращают его, а ты передаешь на выход контроллера то же, что поступало на вход

Anakael replied:
Нет, они там не получают результат сервиса. Я отдаю тоже самое + Id'шник.

Anakael replied:
Done.

arlekin147 commented on TreasuryDepartment/Models/Friend.cs:
```diff
@@ -1,16 +1,11 @@
 namespace TreasuryDepartment.Models
 {
-    public class Friend
+    public class Friend : Offer
```
Друг наследуется от офера? Это точно так и должно быть?

Anakael replied:
Да. Это было сделано для того, чтобы избавиться от сущности Invite.

arlekin147 replied:
А, понял. Меня просто ввело в заблуждение название класса. Может переименуем его во что-то типа FriendOffer? А то возникает чувство, что это прям ДРУГ

Anakael replied:
Это потенциальный друг. Со стороны базы данных они отличаюьтся одним полем. 
Offer - это базовая сущность как для друзей, там и для баланса, так и для сделки. У них у всех есть общие поля (2 пользователя и статус).

arlekin147 replied:
Это все я понял. Я к тому, что класс обозначает, таки, не друга, а предложение о дружбе

Anakael replied:
Ну, посколько мы нигде друга непосредственно не отдаём - можно везде переименовать во FriendInvite

arlekin147 replied:
Я собственно это и предложил

Anakael replied:
Done.

Anakael committed:
Change UserController and Service return type

Merge branch 'release/0.1.0' of github.com:Anakael/Treasury-Department into feature/core/friendservice

# Conflicts:
#	TreasuryDepartment/Models/BlackList.cs
#	TreasuryDepartment/Models/Friend.cs
#	TreasuryDepartment/Models/TreasuryDepartmentContext.cs

Anakael committed:
Add frinend service and route

Merge branch 'release/0.1.0' of github.com:Anakael/Treasury-Department into feature/core/friendservice

# Conflicts:
#	TreasuryDepartment/Models/BlackList.cs
#	TreasuryDepartment/Models/Friend.cs
#	TreasuryDepartment/Models/TreasuryDepartmentContext.cs
